### ISSUE TITLE
Remove dependency constraint on active_record-acts_as

### ISSUE THREAD

**Issue by weiqingtoh** [Coursemology/coursemology2#1541]
Arising from #1512, there is a constraint on `active_record-acts_as` to not update beyond `v2`. 

Updating of `active_record-acts_as` resulted in the following error:
```
NoMethodError: undefined method `polymorphic?' for nil:NilClass
```

> Have spent some time investigating this bug, offending line seems to be [here](https://github.com/lowjoel/schema_plus_association_inverses/blob/master/lib/schema_plus/association_inverses/middleware/association.rb#L13). This is part of the `schema_plus_association_inverses` gem. 
> 
> ```shell
> $ env
> <KeyStruct.accessor:0x3ff34b551058 model:Course::Condition::Achievement(id: integer, achievement_id: integer) name::achievement scope:{:class_name=>"Course::Achievement", :inverse_of=>:achievement_conditions} options:{} extension:nil … >
> $ env.model
> => Course::Condition::Achievement(id: integer, achievement_id: integer)
> $ env.name
> => :achievement
> $ env.model.reflect_on_all_associations.map(&:name)
> => [:creator, :updater, :actable, :course, :conditional, :condition]
> $ env.results.keys
> => ["creator", "updater", "actable", "course", "conditional", "condition", "achievement"]
> ```
> 
> By right `env.model.reflect_on_all_associations.map(&:name)` should include `achievement`, but for some reason it is not being included, more investigation required. 

Further comments
> Did more research, offending gem is not `schema_plus_association_inverses`, but rather `active_record-acts_as`. 
> 
> After testing on my local gem copy, offending commit is https://github.com/hzamani/active_record-acts_as/commit/e61ea582ba898e335d2e344b054f69fd93edd84d. 
> 
> This commit was because rails will be deprecating `alias_method_chain` in favour of `prepend`. See [this article](http://www.justinweiss.com/articles/rails-5-module-number-prepend-and-the-end-of-alias-method-chain/) for a quick summary.
> 
> Will need time to dig into the internals to figure out what actually went wrong. For now, patched gemfile to not update gem to v2.0.

**Comment by weiqingtoh** [comment]
For records, removing this line which 'caches' `_reflections` would solve the problem. However, still unable to find out the root cause. 

One hypothesis is that `schema_plus_core` / `schema_monkey` does not sit well with `active_record-acts_as`. Removing `schema_plus_association_inverses` gem also results in missing associations. See following error message:

```
ArgumentError: No association found for name `options'. Has it been defined yet?
  /Users/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/activerecord-4.2.7.1/lib/active_record/nested_attributes.rb:319:in `block in accepts_nested_attributes_for'
  /Users/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/activerecord-4.2.7.1/lib/active_record/nested_attributes.rb:307:in `each'
  /Users/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/activerecord-4.2.7.1/lib/active_record/nested_attributes.rb:307:in `accepts_nested_attributes_for'
  /Users/Repo/coursemology2/app/models/course/assessment/question/multiple_response.rb:12:in `<class:MultipleResponse>'
  /Users/Repo/coursemology2/app/models/course/assessment/question/multiple_response.rb:2:in `<top (required)>'
  /Users/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/dependencies.rb:274:in `require'
```

To investigate further.

**Comment by allenwq** [comment]
Fixed in #1603
